### PR TITLE
Add path aliases

### PR DESCRIPTION
**Pull request by lukasoppermann**
## Summary

I added path aliases to avoid using relative paths like `../../src/test-utilities/` and make it easier to experiment with different directory structures.

```js
"~types/*": ["types/*"],
"~build-config/*": ["config/*"],
"~test-utilities/*": ["src/test-utilities/*"]
```

## List of notable changes:

- **added** path aliases to `tsconfig.json`
- **updated** build script to register paths
- **updated** jest config to use paths
- **updated** paths in all files

## What should reviewers focus on?

- changes make sense
- any idea why this config https://github.com/primer/primitives/blob/53f9e15d85ea4f76fd0e5997b43755d48e7da733/tsconfig.json for `ts-node` to use aliases does not work?
It is recommended: https://typestrong.org/ts-node/docs/paths/ but it only works for me with `ts-node -r tsconfig-paths/register` 😢 

## Steps to test:

1. Clone branch
1. `yarn`
1. check out files or run `npm t` or `npm run build:color-tokens`

## Supporting resources (related issues, external links, etc):
- https://typestrong.org/ts-node/docs/paths/

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change
- [ ] Verify the design tokens changed in this PR are expected using the diffing results in this PR

### CODE REVIEW

**Review by rezrah — APPROVED** [review]
This looks good and seems to work as expected. 

I've left some additional feedback for you. Strong opinions weekly held though.

File: package.json [hunk 1]
```diff
@@ -91,6 +91,7 @@
     "style-dictionary": "3.7.1",
     "ts-jest": "^29.0.3",
     "ts-node": "^8.10.2",
+    "tsconfig-paths": "^4.1.0",
```
Comment (rezrah) [inline package.json:94]: Is this specifically needed for the node_module imports?

File: tsconfig.json [hunk 2]
```diff
@@ -11,8 +11,17 @@
     "typeRoots": ["./src/@types", "./node_modules/@types", "types", "config/types/"],
     "rootDir": "./",
     "declaration": true,
-    "resolveJsonModule": true
+    "resolveJsonModule": true,
+    "baseUrl": ".",
+    "paths": {
+      "~types/*": ["types/*"],
+      "~build-config/*": ["config/*"],
+      "~test-utilities/*": ["src/test-utilities/*"]
```
[the remark this answers is not in the document]
Comment (lukasoppermann) [reply tsconfig.json:19]: So you are suggesting to just do:

```json
"paths": {
    "~/*": ["./*"]
 },
```

File: package.json (hunk 1 above)
Comment (lukasoppermann) [reply package.json:94]: No, for any import with aliases when running `ts-node`

File: tsconfig.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (rezrah) [reply tsconfig.json]: yep... what do you think of that approach?

[the remark this answers is not in the document]
Comment (lukasoppermann) [reply tsconfig.json]: Just pushed it. 👍

File: package.json (hunk 1 above)
Comment (rezrah) [reply package.json:94]: ah okay, I hadn't run into this in the past.. good to know. Bit annoying it doesn't just respect the paths settings in tsconfig 😞 .

**Pull request merged**